### PR TITLE
backend change for supporting new survival data format

### LINKED ISSUE #7335: (backend and pipeline) Modify importer to convert the current survival data to 0/1
Modify importer to convert the current data to 0/1:

Need to do the following:
- [x] change `importClinicalData.java` importer to import 0/1 survival data
- [x] update related unit tests (importer)
- [x] change data validator to validate 0/1 survival data
- [x] update related unit tests (validator)
- [x] change all old reference in the backend
- [x] update related DB test
- [x] add validation for all survival data

Update survival data (add `0:` or `1:` before the current survival status data):
- [x] (remove) change of `importClinicalData.java` importer
- [x] (remove) change of update related unit tests (importer)
- [x] update data validator to validate survival data with new `0:` or `1:`
- [x] update related unit tests (validator)
- [x] update all old reference in the backend
- [x] update related DB test
- [x] update validation for all survival data

### PR DESCRIPTION
**Pull request by dippindots**
Fix #7335 

This pr will do the following thing:

1. For OS, DFS data, we still allow the data to be imported as old format (“DECEASED”, “LIVING”) importer will import them as new format. The validator will not complain about the OS and DFS to not have “0:” or “1:” before the data.
2. For other survival status attributes, the validator will need them to be the new format or need to run the survival data migration tool.
3. Changed all existing tests data file to new data format
4. Changed all reference (although some of them are already outdated: like 5. GetSurvivalDataJSON.java and clinical_timeline.jsp )
5. Update all MD files to the new format.

### CODE REVIEW

File: core/src/main/scripts/importer/validateData.py [hunk 1]
```diff
@@ -2655,30 +2655,51 @@ def checkLine(self, data):
                                    'column_number': col_index + 1,
                                    'cause': value})
             elif col_name == 'OS_STATUS':
-                if value == 'DECEASED':
+                if value == '1:DECEASED':
                     osstatus_is_deceased = True
                 elif (value.lower() not in self.NULL_VALUES and
-                        value not in ('LIVING', 'DECEASED')):
+                        value not in ('LIVING', 'DECEASED', '0:LIVING', '1:DECEASED')):
```
Comment (n1zea144) [inline core/src/main/scripts/importer/validateData.py:2661]: Should we pull these values out and put into a file for reading into the validator?  I thought we discussed something like this during the planning call.

File: docs/File-Formats.md [hunk 2]
```diff
@@ -202,8 +202,8 @@ Custom attributes:
 #STRING<TAB>STRING<TAB>NUMBER<TAB>STRING<TAB>NUMBER<TAB>...
 #1<TAB>1<TAB>1<TAB>1<TAB>1<TAB>
 PATIENT_ID<TAB>OS_STATUS<TAB>OS_MONTHS<TAB>DFS_STATUS<TAB>DFS_MONTHS<TAB>...
-PATIENT_ID_1<TAB>DECEASED<TAB>17.97<TAB>Recurred/Progressed<TAB>30.98<TAB>...
-PATIENT_ID_2<TAB>LIVING<TAB>63.01<TAB>DiseaseFree<TAB>63.01<TAB>...
+PATIENT_ID_1<TAB>1<TAB>17.97<TAB>1<TAB>30.98<TAB>...
```
Comment (n1zea144) [inline docs/File-Formats.md:205]: Are we missing ":DECEASED" "1:Recurred/Progressed"?

File: portal/src/main/webapp/WEB-INF/jsp/patient_view/clinical_timeline.jsp [hunk 3]
```diff
@@ -159,7 +159,7 @@
 
                 for (i=0; i < prefixes.length; i++) {
                     var prefix = prefixes[i];
-                    if (patientInfo[prefix+"OS_STATUS"] === "DECEASED" &&
+                    if (patientInfo[prefix+"OS_STATUS"] === "1:DECEASED" &&
```
Comment (n1zea144) [inline portal/src/main/webapp/WEB-INF/jsp/patient_view/clinical_timeline.jsp:162]: Are we not generaIizing for other survival data here?

File: portal/src/main/webapp/css/patient-view/clinical-attributes.css [hunk 4]
```diff
@@ -120,15 +120,15 @@
     content: " years old";
 }
 /* attributes with special colors */
-.clinical-attribute[attr-id="OS_STATUS"][attr-value="DECEASED"],
-.clinical-attribute[attr-id="OS_STATUS"][attr-value="DEAD"],
+.clinical-attribute[attr-id="OS_STATUS"][attr-value="1:DECEASED"],
```
Comment (n1zea144) [inline portal/src/main/webapp/css/patient-view/clinical-attributes.css:123]: Are we not generalizing for other survival types here (admittedly I don't know where this file is used).

File: core/src/main/java/org/mskcc/cbio/portal/servlet/GetSurvivalDataJSON.java [hunk 5]
```diff
@@ -144,9 +144,9 @@ protected void doPost(HttpServletRequest httpServletRequest,
                     String osStatus = clinicalData.getOverallSurvivalStatus();
                     if(osStatus == null || osStatus.length() == 0) {
                         _result.put("status", "NA");
-                    } else if (osStatus.equalsIgnoreCase("DECEASED")) {
+                    } else if (osStatus.equalsIgnoreCase("1:DECEASED")) {
```
[the remark this answers is not in the document]
Comment (dippindots) [reply core/src/main/java/org/mskcc/cbio/portal/servlet/GetSurvivalDataJSON.java:147]: I think this file is outdated. I just want to keep the consistency between all the files, so I also changed this.

File: core/src/main/scripts/importer/validateData.py (hunk 1 above)
Comment (dippindots) [reply core/src/main/scripts/importer/validateData.py:2661]: Do you mean for survival data migration tool? As Ramya suggested, I will put the vocabularies in a separate file for the migration tool. I can't really remember we have discussion about this... (please remind me if I missed something...)

File: portal/src/main/webapp/WEB-INF/jsp/patient_view/clinical_timeline.jsp (hunk 3 above)
Comment (dippindots) [reply portal/src/main/webapp/WEB-INF/jsp/patient_view/clinical_timeline.jsp:162]: I think this file is also outdated. We are not using this anymore. I just want to keep the consistency between all the files, so I also changed this.

File: portal/src/main/webapp/css/patient-view/clinical-attributes.css (hunk 4 above)
Comment (dippindots) [reply portal/src/main/webapp/css/patient-view/clinical-attributes.css:123]: Same here, I think this file is also outdated. We are not using this anymore. I just want to keep the consistency between all the files, so I also changed this.

File: docs/File-Formats.md (hunk 2 above)
Comment (dippindots) [reply docs/File-Formats.md:205]: Thanks, I forgot to update this table, will update this now.

File: core/src/main/java/org/mskcc/cbio/portal/servlet/GetSurvivalDataJSON.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (n1zea144) [reply core/src/main/java/org/mskcc/cbio/portal/servlet/GetSurvivalDataJSON.java:147]: I think it is still used by the R and MATLAB libraries, so its good to keep it consistent.

File: core/src/main/java/org/mskcc/cbio/portal/scripts/ImportClinicalData.java [hunk 6]
```diff
@@ -559,7 +558,12 @@ private void addDatum(int internalId, String attrId, String attrVal, String attr
         // attribute value exists and if so, perfom an update
         if (attrType.equals(ClinicalAttribute.PATIENT_ATTRIBUTE)) {
             numPatientSpecificClinicalAttributesAdded++;
-            DaoClinicalData.addPatientDatum(internalId, attrId, attrVal.trim());
+            // add prefix for OS or DFS attributes
+            if (SurvivalAttributeUtil.isSurvivalAttribute(attrId)) {
```
Comment (n1zea144) [inline core/src/main/java/org/mskcc/cbio/portal/scripts/ImportClinicalData.java:562]: I suggest using the "has" approach as implemented in ImportClinicalData. MissingAttributeValues.  It makes for cleaner code - 

https://github.com/cBioPortal/cbioportal/blob/b69759630a61eb0040436987801ad57079cfdcfb/core/src/main/java/org/mskcc/cbio/portal/scripts/ImportClinicalData.java#L81

File: core/src/main/java/org/mskcc/cbio/portal/servlet/GetSurvivalDataJSON.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (n1zea144) [reply core/src/main/java/org/mskcc/cbio/portal/servlet/GetSurvivalDataJSON.java:147]: I think this code may still be used by the old web service (R and MATLAB libraries included).  If I understand the code, you are transforming a value of "1:DECEASED" to "1" on the way out of this servlet...but why do you want to do this?

File: core/src/main/java/org/mskcc/cbio/portal/util/SurvivalAttributeUtil.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+        }
+        for (SurvivalStatusMappingZeroValues attributeValue : SurvivalStatusMappingZeroValues.values()) {
+            if (attributeValue.getStatusValue().equals(value)) {
+                return ZERO_PREFIX + value;
+            }
+        }
+        // value is new format or is one of the NULL_VALUES
+        return value;
+    }
+}
```
Comment (n1zea144) [inline core/src/main/java/org/mskcc/cbio/portal/util/SurvivalAttributeUtil.java:75]: Please add a newline here.

File: core/src/main/java/org/mskcc/cbio/portal/util/SurvivalAttributeUtil.java [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+            statusValue = value;
+        }
+
+        String getStatusValue() {
+            return statusValue;
+        }
+    }
+
+    public static boolean isSurvivalAttribute(String attributeId) {
+        for (SurvivalStatusAttributes attribute : SurvivalStatusAttributes.values()) {
```
Comment (n1zea144) [inline core/src/main/java/org/mskcc/cbio/portal/util/SurvivalAttributeUtil.java:53]: Suggest using the "has" approach as linked above...

File: core/src/main/java/org/mskcc/cbio/portal/util/SurvivalAttributeUtil.java [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+            if (attribute.getAttributeId().equals(attributeId)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static String getModifiedAttributeValue(String value) {
```
Comment (n1zea144) [inline core/src/main/java/org/mskcc/cbio/portal/util/SurvivalAttributeUtil.java:61]: Suggest using the "has" approach as linked above...

File: core/src/main/scripts/importer/validateData.py (hunk 1 above)
Comment (n1zea144) [reply core/src/main/scripts/importer/validateData.py:2661]: You remember correctly, we did discuss as it relates to survival data migration tool.  I'm suggesting we use a similar approach to inject the vocabulary here.  Maybe you are only validating OS_STATUS and DFS_STATUS and nothing more?

Comment (dippindots) [reply core/src/main/scripts/importer/validateData.py:2661]: Yes, we are just checking the `OS_STATUS` and `DFS_STATUS` here, for other values. For other status, we require all the values have the prefix `0:` and `1:` https://github.com/cBioPortal/cbioportal/blob/86f066aca39614fc7d7978e746d95d391b34162c/core/src/main/scripts/importer/validateData.py#L2688-L2702

**Comment by inodb** [comment]
Just gonna merge these survival migration related PRs to release-3.4.0 branch. Makes it a little easier to deploy to the beta env

**Pull request merged**